src/Microsoft.AspNet.Mvc.DataAnnotations/CompareAttributeAdapter.cs

```diff
@@ -28,65 +27,43 @@ public CompareAttributeAdapter(CompareAttribute attribute, IStringLocalizer stri
                 throw new ArgumentNullException(nameof(context));
             }
 
-            var errorMessage = ((CompareAttributeWrapper)Attribute).FormatErrorMessage(context);
+            var errorMessage = GetErrorMessage(context);
             var clientRule = new ModelClientValidationEqualToRule(errorMessage,
                                                             FormatPropertyForClientValidation(Attribute.OtherProperty));
             return new[] { clientRule };
         }
 
-        private static string FormatPropertyForClientValidation(string property)
+        /// <inheritdoc />
+        public override string GetErrorMessage(ModelValidationContextBase validationContext)
```

Comment:
Oh wait, nm, if `otherPropertyDisplayName != null` this is ok to be `null`. GOod as is